### DISCUSSION TITLE
How do I update materials automatically in case of failed retry job

### DISCUSSION THREAD

**Discussion by xlw712** [gocd/gocd#11411] [Q&A]
How do I update materials automatically in case of failed retry job?
I'm now using the following interface
```
$ curl 'https://ci.example.com/go/api/stages/myPipeline/myPipelineRunCounter/myStage/myStageCounter/run-failed-jobs' \
      -u 'username:password' \
      -H 'X-GoCD-Confirm: true' \
      -H 'Accept: application/vnd.go.cd.v3+json' \
      -X POST
```

**Comment by xlw712** [comment]
I just have scripts in my materials that I want to update every time

**Comment by xlw712** [comment]
Is there any good way

**Comment by chadlwilson** [comment]
It's not clear what the problem that you have is, or what you are trying to achieve from the above, so not sure how I can help.

**Reply by xlw712** [reply]
> It's not clear what the problem that you have is, or what you are trying to achieve from the above, so not sure how I can help.

emm, I want the interface to retry and update materials automatically when the job in the pipeline fails？

**Reply by chadlwilson** [reply]
When you 're-run' jobs within a pipeline stage you will always do so with the same materials as the jobs were originally run with. That's a core guarantee GoCD offers.

If you want GoCD to check the materials for updates again you need to schedule/trigger the entire pipeline, not re-run jobs.

Otherwise some jobs in a stage would have been run with material version X while others run with material version Y which would violate core principles of GoCD and build/job reproducibility.

If you want to automate it externally you could use the APIs to find pipelines with failed jobs and then call the pipeline 'schedule' API to re-schedule the entire pipeline (and all stages/jobs within it). Unless you tell the API specific material revisions it will check and run on the latest material revisions.
